# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/power/WakeLockSentinel.h
# Commit: f50ff29a1a64
# Full Hash: f50ff29a1a649e3aad723b592bdd2c3c325b3d99
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-06 05:18:37
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 1: Screen Wake Lock boilerplate. r=dom-core,smaug,emilio,edgar
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189508
# ==============================================================================

diff -r 00afe2e46b1b -r f50ff29a1a64 dom/power/WakeLockSentinel.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/power/WakeLockSentinel.h	Tue Dec 05 23:58:06 2023 +0000
@@ -0,0 +1,49 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim:set ts=2 sw=2 sts=2 et cindent: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef DOM_WAKELOCKSENTINEL_H_
+#define DOM_WAKELOCKSENTINEL_H_
+
+#include "js/TypeDecls.h"
+#include "mozilla/DOMEventTargetHelper.h"
+#include "mozilla/dom/WakeLockBinding.h"
+
+namespace mozilla::dom {
+
+class Promise;
+
+}  // namespace mozilla::dom
+
+namespace mozilla::dom {
+
+class WakeLockSentinel final : public DOMEventTargetHelper {
+ public:
+  WakeLockSentinel(nsIGlobalObject* aOwnerWindow, WakeLockType aType)
+      : DOMEventTargetHelper(aOwnerWindow), mType(aType) {}
+
+ protected:
+  ~WakeLockSentinel() = default;
+
+ public:
+  JSObject* WrapObject(JSContext* aCx,
+                       JS::Handle<JSObject*> aGivenProto) override;
+
+  bool Released() const;
+
+  WakeLockType Type() const { return mType; }
+
+  MOZ_CAN_RUN_SCRIPT
+  already_AddRefed<Promise> ReleaseLock();
+
+  IMPL_EVENT_HANDLER(release);
+
+ private:
+  WakeLockType mType;
+};
+
+}  // namespace mozilla::dom
+
+#endif  // DOM_WAKELOCKSENTINEL_H_